server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -712,9 +712,10 @@ def _analyze(
         return out
 
     def _agg(self, samples: np.ndarray, offsets: np.ndarray) -> float:
-        lengths = np.diff(offsets.base)
+        lengths = np.diff(offsets.base).astype(float)
+        lengths[lengths == 0] = np.nan
         vec = np.add.reduceat(samples, offsets) / lengths
-        vec[lengths == 0] = 0  # fix NaNs + reduceat copies the previous value
+        vec[np.isnan(vec)] = 0  # fix NaNs + reduceat copies the previous value
```

Comment:
Why doesn't the original mask work? This is important because the new code is less efficient.

Reply:
tried with a simple mask